influxdb/client.py

```diff
@@ -342,6 +354,21 @@ def _read_chunked_response(response, raise_errors=True):
                             _key, []).extend(result[_key])
         return ResultSet(result_set, raise_errors=raise_errors)
 
+    @staticmethod
+    def _parse_msgpack(response):
+        """Return the decoded response if it is encoded as msgpack."""
+        def hook(code, data):
+            if code == 5:
+                (epoch_s, epoch_ns) = struct.unpack(">QI", data)
+                time = datetime.datetime.utcfromtimestamp(epoch_s)
+                time += datetime.timedelta(microseconds=(epoch_ns / 1000))
+                return time.isoformat() + 'Z'
+            return msgpack.ExtType(code, data)
+
+        headers = response.headers
```

Comment:
I would prefer if this logic wasn't in parse_msgpack.

I think It would be easier to read if checking the headers was in query()

Reply:
Did you mean in `request` ?

Reply:
Hmm yes sorry. What do you think?

Reply:
Fixed in the latest commits